ops/cli.sh

```diff
@@ -0,0 +1,166 @@
+error   () { >&2 printf '\e[1;91m%-6s\e[m\n' "$*"; } # red
+success () {     printf '\e[1;92m%-6s\e[m\n' "$*"; } # green
+warn    () {     printf '\e[1;93m%-6s\e[m\n' "$*"; } # yellow
+debug   () {                                         # gray
+    test "$DEBUG" = true && printf '\e[1;90m%-6s\e[m\n' "$*"
+}
+
+# Draw a line of - across the width of the terminal.
+#
+line () { printf "%.$(tput cols)d\n" 0 | tr 0 -; }
+
+
+# OK when each CMD is on the PATH.
+#
+function is_available () {
+    local cmd ok=ok
+    for cmd in "$@"
+    do
+        if 2>/dev/null hash "$cmd"
+        then
+            : OK, we have "$cmd"
+        else
+            unset ok
+            error "$cmd" is required but not found!
+            info "${INSTALL_MSG}"
+        fi
+    done
+    test "$ok"
+}
+
+# Wait at least 10 seconds for a result from "@".
+#
+wait_for_result () {
+    local n result
+    for n in 0 1 2 3 4 5 6 7 8 9
+    do
+        sleep 1
+        result=$("$@")
+        test "$result" && echo "$result" && break
+    done
+    test "$result"
+}
+
+# Render Helm values to CTMPL_FILE.
+#
+function run_render () {
+    local -r ctmpl_file=$1
+    if test "$ctmpl_file"
+    then
+        info Rendering "$ctmpl_file" for you...
+        docker run -i \
+               --rm \
+               -v "$(pwd)":/working \
+               -v "$HOME"/.vault-token:/root/.vault-token \
+               broadinstitute/dsde-toolbox:dev \
+               /usr/local/bin/render-ctmpls.sh \
+               -k "$ctmpl_file"
+    else
+        error A valid path to ctmpl file is required!
+        info "$HELP_MSG"
+    fi
+}
+
+function setup_and_update_helm_charts () {
+    local -r repo=${1:-gotc-charts}
+    helm repo add "$repo" https://broadinstitute.github.io/gotc-helm-repo/
+    helm repo update
+    helm repo list
+}
+
+# Install or upgrade DEPLOYMENT, then list.
+# Enable ingress when when INGRESS is set.
+#
+function deploy_helm_charts () {
+    local -r deployment=$1 repo=$2 chart=$3 rendered=$4 ingress=$5
+    local -a upgrade=(helm upgrade "$deployment" "$repo/$chart"
+                      -f "$rendered" --install)
+    test "$ingress" || upgrade+=(--set ingress.enabled=false)
+    "${upgrade[@]}"
+    helm list
+}
+
+# Deploy on the gotc-dev shared cluster.
+#
+function run_deploy () {
+    local -r ctx=gke_broad-gotc-dev_us-central1-a_gotc-dev-shared-us-central1-a
+    kubectl config use-context $ctx
+    run_render wfl-values.yaml.ctmpl
+    setup_and_update_helm_charts gotc-charts
+    deploy_helm_charts gotc-dev gotc-charts wfl wfl-values.yaml ingress
+}
+
+# Deploy to a local minikube and forward the pod to PORT.
+#
+function run_local () {
+    local -r port=18982
+    local -ar getpods=(kubectl get pods --no-headers
+                       -o custom-columns=:metadata.name)
+    line
+    # minikube start is not idempotent, kill
+    # all previous ones for the sake of your
+    # laptop's safety
+    minikube stop
```

Comment:
I think this should be the responsibility of whoever is running the test. Stopping minikube globally might be annoying if people are working on multiple things simultaneously.
Can we use setup/teardown fixture semantics for this to isolate this deployment from others?